webplugin/js/app/mck-app.js

```diff
@@ -321,10 +327,10 @@ function ApplozicSidebox() {
 
             // Remove scripts if chatwidget is restricted by domains
             var isCurrentDomainDisabled = Array.isArray(allowedDomains) && allowedDomains.length && !allowedDomains.some(isSubDomain);
-
+            var isCurrentDomainKommunicate = kommunicateDomains.some(isSubDomain);
             // Remove scripts if disableChatWidget property is enabled
             // or domain restrictions are enabled
-            if (disableChatWidget || isCurrentDomainDisabled) {
+            if ((disableChatWidget || isCurrentDomainDisabled) && !isCurrentDomainKommunicate) {
```

Comment:
@shubhamtibra You can concat two arrays.
1. Array which is coming from widget settings 
2. Array which you have hard coded here.

You won't need an extra variable and check then.
NOTE: spread operator and object.assign doesn't work in IE. So to merge two arrays, use [array concat](https://www.w3schools.com/Jsref/jsref_concat_array.asp) or some other thing which works in IE.

Reply:
>  var isCurrentDomainDisabled = Array.isArray(allowedDomains) && allowedDomains.length && !allowedDomains.some(isSubDomain);

Allowed Domains might be empty array which means the setting is disabled. If we concat the two, we wouldn't be able to check this condition.

Reply:
Good point, if the array is not empty then you can contact, otherwise no need to merge the arrays as it will be allowed on every domain.

Reply:
The condition to check if domain is allowed and domain is kommunicate needs to be different.
Because if the domain is kommunicate we allow in both cases.

Reply:
If i put them in a single array I will get a single result but I need the kommunicate result to allow in disabled settings too.

Reply:
Sample code:
```
        var isCurrentDomainDisabled;
        if (Array.isArray(allowedDomains) && allowedDomains.length) {
            allowedDomains = allowedDomains.concat(["kommunciate.io"]); // add your concat logic here 
            isCurrentDomainDisabled = !allowedDomains.some(isSubDomain);
        }
```
If something is failing in the above code, ping me over slack.
Thanks.